lib/galaxy/datatypes/constructive_solid_geometry.py

```diff
@@ -813,3 +815,63 @@ def get_next_line(fh):
         # Discard the rest of the line
         fh.readline()
     return line.strip()
+
+class Vtu(GenericXml):
+    """Format for defining VTU (VTK Unstructured Grid) data https://docs.vtk.org/"""
+
+    edam_data = "edam:data_3671"
+    edam_format = "edam:format_3621"
+    file_ext = "vtu"
+
+    MetadataElement(name="file_format", default=None, desc="File format", readonly=True, optional=True, visible=True)
+    MetadataElement(name="num_cells", desc="Number of cells in the mesh", param_type="int", optional=True)
+    MetadataElement(name="num_points", desc="Number of points in the mesh", param_type="int", optional=True)
+
+    def set_meta(self, dataset: DatasetProtocol, **kwd) -> None:
+        """Set metadata for VTU files, including number of cells and points."""
+        try:
+            with open(dataset.file_name, "r") as file:
+                tree = ET.parse(file)
```

Comment:
Wondering if it's really worth to read the complete file or we restrict the metadata to what we can get from the attributes of the root XML tag (i.e. the first line of the file). This should be `vtk_version` and `dataset_type`. 

@bgruening do you know a too that uses the "legacy" `vtk` datatype? Then we could check if additional metadata could be helpful.